File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    isVPS = True
    
    for i in s:
        if i == '(':
            stack.append(i)
        elif i == ')':
            if stack:
                stack.pop()
            else:
                isVPS = False
                break
    if not stack and isVPS:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let count = 0;
    
    if (s[0] === ")") return false;
    
    for (const char of s) {
        if (char === "(") count++;
        else count--;
        
        if (count < 0) return false;
    }

    return count === 0;
}